mongo/mongo.go

```diff
@@ -129,33 +129,45 @@ func createImageUpdateQuery(ctx context.Context, id string, image *models.Image)
 
 	log.Event(ctx, "building update query for image resource", log.INFO, log.INFO, log.Data{"image_id": id, "image": image, "updates": updates})
 
-	if image.CollectionID != "" {
-		updates["next.collection_id"] = image.CollectionID
-	}
-
-	if image.State != "" {
-		updates["next.state"] = image.State
-	}
-
-	if image.Filename != "" {
-		updates["next.filename"] = image.Filename
-	}
+	updates["collection_id"] = image.CollectionID
+	updates["state"] = image.State
+	updates["filename"] = image.Filename
+	updates["type"] = image.Type
 
 	if image.License != nil {
-		updates["next.license.title"] = image.License.Title
-		updates["next.license.href"] = image.License.Href
+		license := make(bson.M)
+		license["title"] = image.License.Title
+		license["href"] = image.License.Href
+		updates["license"] = license
+	} else {
+		updates["license"] = nil
 	}
 
 	if image.Upload != nil {
-		updates["next.upload.path"] = image.Upload.Path
-	}
-
-	if image.Type != "" {
-		updates["next.type"] = image.Type
+		upload := make(bson.M)
+		upload["path"] = image.Upload.Path
+		updates["upload"] = upload
+	} else {
+		updates["upload"] = nil
 	}
 
 	if image.Downloads != nil {
-		updates["next.downloads"] = image.Downloads
+		k0Downloads := make(bson.M)
```

Comment:
not sure about the variable names k0Downloads and K1Downloads, what are these?

Reply:
Confused me for a bit too. It's the double keys of the downloads map. Ie. k0 is "png" and "k1" is "1024x768" etc.

Reply:
If we just need to allow overwriting single fields and keeping the rest, I will leave it as it was.

Reply:
So would it be better to refer to k0Downloads as imageDownloads and k1Downloads as resolutionDownloads? Just trying to be more explicit and less cryptic?

Reply:
I went for `variants` and `resolutions`